<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Widget</string>
  </property>
  <widget class="QWidget" name="verticalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>40</x>
     <y>170</y>
     <width>341</width>
     <height>316</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QGridLayout" name="puzzleGrid">
      <item row="2" column="1">
       <widget class="QPushButton" name="pushButton_8">
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QPushButton" name="pushButton_9">
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QPushButton" name="pushButton_4">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QPushButton" name="pushButton_2">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="pushButton_5">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QPushButton" name="pushButton_3">
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QPushButton" name="pushButton_7">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="pushButton_6">
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QPushButton" name="pushButton_1">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">font: 14pt &quot;阿里妈妈东方大楷&quot;;</string>
        </property>
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QPushButton" name="initButton">
   <property name="geometry">
    <rect>
     <x>50</x>
     <y>510</y>
     <width>141</width>
     <height>41</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color: rgb(255, 188, 140);
font: 16pt &quot;Calibri&quot;;</string>
   </property>
   <property name="text">
    <string>Initialization</string>
   </property>
  </widget>
  <widget class="QPushButton" name="runButton">
   <property name="geometry">
    <rect>
     <x>230</x>
     <y>510</y>
     <width>141</width>
     <height>41</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">font: 16pt &quot;Calibri&quot;;
background-color: rgb(255, 188, 140);</string>
   </property>
   <property name="text">
    <string>Run</string>
   </property>
  </widget>
  <widget class="QComboBox" name="algorithmCombo">
   <property name="geometry">
    <rect>
     <x>460</x>
     <y>230</y>
     <width>301</width>
     <height>41</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">
font: 12pt &quot;Calibri&quot;;</string>
   </property>
   <item>
    <property name="text">
     <string>BFS (Breadth-First-Search)</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>DFS (Depth-First-Search)</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>BFS (Best-First-Search)</string>
    </property>
   </item>
   <item>
    <property name="text">
     <string>Branch-And-Bound</string>
    </property>
   </item>
  </widget>
  <widget class="QLabel" name="selectLabel">
   <property name="geometry">
    <rect>
     <x>460</x>
     <y>180</y>
     <width>321</width>
     <height>20</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">font: 20pt &quot;阿里妈妈东方大楷&quot;;</string>
   </property>
   <property name="text">
    <string>Please select one of them:</string>
   </property>
  </widget>
  <widget class="QLabel" name="titleLabel">
   <property name="geometry">
    <rect>
     <x>80</x>
     <y>50</y>
     <width>671</width>
     <height>51</height>
    </rect>
   </property>
   <property name="styleSheet">
    <string notr="true">font: 36pt &quot;阿里妈妈东方大楷&quot;;</string>
   </property>
   <property name="text">
    <string>Solution for 8 - Puzzle problem</string>
   </property>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>460</x>
     <y>430</y>
     <width>221</width>
     <height>91</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QLabel" name="timeLabel">
      <property name="styleSheet">
       <string notr="true">font: 16pt &quot;Calibri&quot;;</string>
      </property>
      <property name="text">
       <string>Time(s): 0</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="stepsLabel">
      <property name="styleSheet">
       <string notr="true">font: 16pt &quot;Calibri&quot;;</string>
      </property>
      <property name="text">
       <string>Steps: 0</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="verticalLayoutWidget_3">
   <property name="geometry">
    <rect>
     <x>460</x>
     <y>310</y>
     <width>331</width>
     <height>101</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout_4">
    <item>
     <widget class="QLabel" name="resultLabel">
      <property name="styleSheet">
       <string notr="true">font: 20pt &quot;阿里妈妈东方大楷&quot;;</string>
      </property>
      <property name="text">
       <string>Result:</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QLabel" name="goalLabel">
      <property name="styleSheet">
       <string notr="true">font: 16pt &quot;Calibri&quot;;</string>
      </property>
      <property name="text">
       <string>(No) Goal node</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
